docs/reference/certificates.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: docwithnav
+title: Kubernetes PKI certificates and requirements
+permalink: /certs/
+approvers:
+- efrost
+---
+
+# Introduction
+
+A tool like [kubeadm][kubeadm] can be used to generate your configuration for you.
+But there are also reasons to generate these certificates yourself.
+This allows you to keep your private keys off of the api server, for enhanced security.
+
+# Certificates in Kubernetes
+
+Kuberentes uses TLS extensively for authentication.
+
+## Where they're used
+
+Kubernetes uses kubeadm for four primary things:
+1. Client certificates for the kubelet to authenticate to the master
+2. Server certificate for the api server endpoint
+3. Client certificates for administrators of the cluster authenticating to the API server
+4. Client certificates for service accounts authenticating to the API server.
```

Comment:
This is being a bit pedantic, but technically the service accounts are using a generated public/private keypair only.

Reply:
Certificates are also used for the following:

* client certificate for the api server to talk to the kubelets
* client certificate for the api server to talk to etcd
* client certificate/kubeconfig for controller manager to talk to api server
* client certificate/kubeconfig for scheduler to talk to api server

Reply:
It may be good to mention the use of the front-proxy certs here as well, which are used to enable extension api servers as described here: https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/